### PR TITLE
feat: use venv on docker image

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

File: device-discovery/docker/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,22 @@
-FROM python:3.12-bullseye
-
-RUN addgroup --system netdev && useradd -m --shell /bin/bash -G netdev appuser && echo "appuser ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+FROM python:3.12-bullseye as builder
```
Comment (mfiedorowicz) [inline device-discovery/docker/Dockerfile]: Use of `as` results with warning:
```
 1 warning found (use docker --debug to expand):
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)
```

```suggestion
FROM python:3.12-bullseye AS builder
```

**Review by mfiedorowicz — COMMENTED** [review]
When building docker image from this Dockerfile, image size is suspiciously large (999MB), is that expected?

**Comment by mfiedorowicz** [comment]
> When building docker image from this Dockerfile, image size is suspiciously large (999MB), is that expected?

FYI: using `python:3.12-slim-bullseye` shrinks image to ~235 MB

**Comment by leoparente** [comment]
> > When building docker image from this Dockerfile, image size is suspiciously large (999MB), is that expected?
> 
> FYI: using `python:3.12-slim-bullseye` shrinks image to ~235 MB

good call @mfiedorowicz 
image

**Review by mfiedorowicz — APPROVED** [review]

**Pull request merged**